apps/api/src/controllers/stateManager.ts

```diff
@@ -63,73 +64,140 @@ export class ApplicationStateManager extends StateMachine<ApplicationStateValues
+			const update = { state };
+			const applicationResult = await applicationRepo.findOneAndUpdate({ id, update });
+
+			if (applicationResult.success && applicationResult.data) {
+				this._application = applicationResult.data;
+			}
+
+			return applicationResult;
+		});
+	}
+
 	// Handler Methods
 	// Submit
+	// TODO: Add Validation + Edit Content service methods
 	async submitDraft() {
-		if (this.can(submit)) {
-			// TODO: Add Validation
+		const transitionResult = this._canPerformAction(submit, ApplicationStates.DRAFT);
+		if (transitionResult.success) {
 			const validationResult = await validateContent(this._application);
 			if (validationResult.success) {
-				await this.dispatch(submit);
-				return validationResult;
+				return await this._dispatchAndUpdateAction(submit, 'draftSubmit');
 			} else {
-				return failure(`Cannot submit application with state ${this.getState()}`);
+				return validationResult;
 			}
 		} else {
-			return failure(`Cannot submit application with state ${this.getState()}`);
+			return transitionResult;
 		}
 	}
 
-	async submitRepReview() {
-		return this.submitDraft();
-	}
-
 	async submitRepRevision() {
-		return this.submitDraft();
+		const transitionResult = this._canPerformAction(submit, ApplicationStates.INSTITUTIONAL_REP_REVISION_REQUESTED);
+		if (transitionResult.success) {
+			return await this._dispatchAndUpdateAction(submit, 'repSubmit');
+		} else {
+			return transitionResult;
+		}
 	}
 
 	async submitDacRevision() {
-		return this.submitDraft();
+		const transitionResult = this._canPerformAction(submit, ApplicationStates.DAC_REVISIONS_REQUESTED);
+		if (transitionResult.success) {
+			return await this._dispatchAndUpdateAction(submit, 'dacSubmit');
+		} else {
+			return transitionResult;
+		}
 	}
 
 	private async _onSubmit() {
 		return success('post dispatch on submit');
 	}
 
 	// Edit
+	// TODO: Add Validation + Edit Content service methods
 	async editDraft() {
-		if (this.can(edit)) {
-			await this.dispatch(edit);
-			return success(edit);
+		const transitionResult = this._canPerformAction(edit, ApplicationStates.DRAFT);
+		if (transitionResult.success) {
+			const validationResult = await validateContent(this._application);
+			return validationResult;
 		} else {
-			return failure(`Cannot edit application with state ${this.getState()}`);
+			return transitionResult;
 		}
 	}
 
 	async editRepReview() {
-		return this.editDraft();
+		const transitionResult = this._canPerformAction(edit, ApplicationStates.INSTITUTIONAL_REP_REVIEW);
+		if (transitionResult.success) {
+			return await this._dispatchAndUpdateAction(edit, 'edit');
+		} else {
+			return transitionResult;
+		}
 	}
 
 	async editDacReview() {
-		return this.editDraft();
+		const transitionResult = this._canPerformAction(edit, ApplicationStates.DAC_REVIEW);
+		if (transitionResult.success) {
+			return await this._dispatchAndUpdateAction(edit, 'edit');
+		} else {
+			return transitionResult;
+		}
 	}
```

Comment:
Edits can be made during Rep Review / Dac Review, but this reverts the application to Draft

This is distinct from Rep Revision / Dac Revision states, where changes are requested and then submitted (submit action vs edit action)

Reply:
@joneubank migrations and 'EDIT' Actions have been removed (closing the old discussion, so I added a comment here)

So for the scope of this PR, the edit actions check that 'edit' event is allowed, but don't create an action or update the application. There's just dummy content validation.